testsuite/cluster-test/src/experiments/twin_validator.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{
+    cluster::Cluster,
+    experiments::{Context, Experiment, ExperimentParam},
+    instance,
+    instance::Instance,
+    stats,
+    tx_emitter::EmitJobRequest,
+    util::unix_timestamp_now,
+};
+use async_trait::async_trait;
+use futures::{future::try_join_all, join};
+use libra_logger::{info, warn};
+use rand::Rng;
+use std::{collections::HashSet, fmt, time::Duration};
+use structopt::StructOpt;
+use tokio::time;
+
+#[derive(StructOpt, Debug)]
+pub struct TwinValidatorsParams {
+    #[structopt(long, default_value = "1", help = "Set twin node pair number")]
+    pub pair: usize,
+}
+
+pub struct TwinValidators {
+    instances: Vec<Instance>,
+    twin_validators: Vec<Instance>,
+}
+
+impl ExperimentParam for TwinValidatorsParams {
+    type E = TwinValidators;
+    fn build(self, cluster: &Cluster) -> Self::E {
+        if self.pair > 3 {
+            panic!("pair number {} can not more than 9", self.pair);
+        }
+        let mut instances = cluster.validator_instances().to_vec();
+        let mut twin_validators = vec![];
+        let mut rnd = rand::thread_rng();
+        for _i in 0..self.pair {
+            twin_validators.push(instances.remove(rnd.gen_range(0, instances.len())));
+        }
+        Self::E {
+            instances,
+            twin_validators,
+        }
+    }
+}
+
+#[async_trait]
+impl Experiment for TwinValidators {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&self.twin_validators)
+    }
+
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        let buffer = Duration::from_secs(60);
+        let window = Duration::from_secs(240);
+        let mut new_instances = vec![];
+        for i in 0..self.twin_validators.len() {
+            let mut new_twin_config = self.twin_validators[i].instance_config().clone();
+            new_twin_config.make_twin(1);
+            let new_inst = context
+                .cluster_swarm
+                .spawn_new_instance(new_twin_config, true)
+                .await?;
+            new_instances.push(new_inst.clone());
+        }
+        let instances = self.instances.clone();
+        let emit_job_request =
+            EmitJobRequest::for_instances(instances, context.global_emit_job_request);
+        let emit_txn = context.tx_emitter.emit_txn_for(window, emit_job_request);
+        let trace_tail = &context.trace_tail;
+        let trace_delay = buffer;
+        let trace = false;
```

Comment:
You don't need trace in this experiment, let's remove it to not make code too complex.